src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,204 @@
+    private Node<T> tail;
+    private int size;
+
+    public Node<T> getHead() {
+        return head;
+    }
+
+    public void setHead(Node<T> head) {
+        this.head = head;
+    }
+
+    public Node<T> getTail() {
+        return tail;
+    }
+
+    public void setTail(Node<T> tail) {
+        this.tail = tail;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node;
+        if (isEmpty()) {
+            node = new Node<>(null, value, null);
+            setHead(node);
+            setTail(node);
+        } else {
+            node = new Node<>(tail, value, null);
+            tail.setNext(node);
+            setTail(node);
+        }
+        setTail(node);
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index
+                    + " out of range " + size());
+        }
+        Node<T> insert;
+        if (index == 0 && getHead() != null) {
+            insert = new Node<>(null, value, getHead());
+            getHead().setPrev(insert);
+            setHead(insert);
+            size++;
+        } else if (getHead() == null || index == size()) {
+            add(value);
+        } else {
+            Node<T> old = getNode(index);
+            insert = new Node<>(old.getPrev(), value, old);
+            old.getPrev().setNext(insert);
+            old.setPrev(insert);
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkRange(index);
+        return getNode(index).getItem();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkRange(index);
+        T oldItem = getNode(index).getItem();
+        getNode(index).setItem(value);
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkRange(index);
+        Node<T> temp = getNode(index);
+        unlink(temp);
+        size--;
+        return temp.getItem();
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> test = head;
+        if (size() == 1) {
+            if (isEquals(object, head.getItem())) {
+                size--;
+                unlink(test);
+                return true;
+            }
+            return false;
+        }
+        while (test.getNext() != null) {
+            if (isEquals(object, test.getItem())) {
+                unlink(test);
+                size--;
+                return true;
+            }
+            test = test.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return this.size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size() == 0;
```

Comment:
you have access to the class variables 
```suggestion
        return size == 0;
```